plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -1276,3 +1288,33 @@ func (r *ReferenceManager) lookupResource(ctx context.Context, resource schema.G
 	}
 	return nil
 }
+
+func hasDecreasedNodeLimits(limits, oldLimits *core.Limits) bool {
+	if limits == nil || apiequality.Semantic.DeepEqual(limits, oldLimits) {
+		// limits have been removed or were not changed.
+		return false
+	}
+	return oldLimits == nil || validation.IsDecreasedMaxNodesTotal(limits.MaxNodesTotal, oldLimits.MaxNodesTotal)
+}
+
+func validateShootWorkerLimits(channel chan error, shoot *gardencorev1beta1.Shoot, limits *core.Limits) {
+	if limits == nil || limits.MaxNodesTotal == nil {
+		return
+	}
+
+	var (
+		maxNodesTotal = *limits.MaxNodesTotal
+		totalMinimum  int32
+	)
+
+	for _, worker := range shoot.Spec.Provider.Workers {
+		totalMinimum += worker.Minimum
+		if worker.Maximum > maxNodesTotal {
+			channel <- fmt.Errorf("the maximum node count of worker pool %q in shoot \"%s/%s\" exceeds the limit of %d total nodes configured in the cloud profile", worker.Name, shoot.Namespace, shoot.Name, maxNodesTotal)
+		}
+	}
+
+	if totalMinimum > maxNodesTotal {
+		channel <- fmt.Errorf("the total minimum node count of all worker pools of shoot \"%s/%s\" must not exceed the limit of %d configured in the cloud profile", shoot.Namespace, shoot.Name, maxNodesTotal)
```

Comment:
Nit: For using the same structure as in the error above
```suggestion
		channel <- fmt.Errorf("the total minimum node count of all worker pools of shoot \"%s/%s\" must not exceed the limit of %d total nodes configured in the cloud profile", shoot.Namespace, shoot.Name, maxNodesTotal)
```